chore: update Node.js version to 22 in configuration files

Puppo opened the pull request:
- Updated .nvmrc, package.json, and package-lock.json to require Node.js version 22.
- Adjusted CI workflow to use Node.js version 22 for consistency.

## Description

Upgrade node version from 18 to 22

## Related Issues

## Check List

- [x] I have read the [Contributing Guidelines](./CONTRIBUTING.md)

luisorbaiceta commented:
@jsumners this solves the issue. I'm leaving this as a draft because I want to check all the edge cases

luisorbaiceta commented:
In fact, these covers more link cases so we can simplify the regexes a lot

jsumners reviewed (approved):
Looks good to me.

I was trying to keep the expressions easier to read. But whatever gets us to ship is cool with me.

luisorbaiceta commented:
I will leave it as is, simplifying regexes could lead to some backwards compatibility issues with previous versions.

luisorbaiceta commented:
> Ship it. ![giphy](https://user-images.githubusercontent.com/321201/83882820-a3582680-a710-11ea-93ee-0f63c7b70a7e.gif)

Seems that I don't write access @mcollina

Puppo committed:
chore: update Node.js version to 22 in configuration files

- Updated .nvmrc, package.json, and package-lock.json to require Node.js version 22.
- Adjusted CI workflow to use Node.js version 22 for consistency.
